x-pack/plugins/integration_assistant/public/components/create_integration/create_integration_assistant/steps/data_stream_step/generation_modal.tsx

```diff
@@ -171,18 +171,28 @@ export const useGeneration = ({
         onComplete(result);
       } catch (e) {
         if (abortController.signal.aborted) return;
-        const errorMessage = `${e.message}${
+        const originalErrorMessage = `${e.message}${
           e.body ? ` (${e.body.statusCode}): ${e.body.message}` : ''
         }`;
 
         reportGenerationComplete({
           connector,
           integrationSettings,
           durationMs: Date.now() - generationStartedAt,
-          error: errorMessage,
+          error: originalErrorMessage,
         });
 
-        setError(errorMessage);
+        const errorCode = e.body?.attributes?.errorCode;
+        let errorMessage;
+        if (errorCode != null) {
+          errorMessage = i18n.ERROR_TRANSLATION[errorCode];
+        }
+
+        if (errorMessage != null) {
+          setError(errorMessage);
+        } else {
+          setError(originalErrorMessage);
+        }
```

Comment:
Using the `enum` mentioned in the previous comment, we can ensure all the error codes have a translation in the object with: 
```typescript
export const ERROR_TRANSLATION: Record<ErrorCode, string> = {
  [ErrorCode.recursionLimit]: i18n.translate(...),
  [ErrorCode.recursionLimitAnalyzeLogs]: i18n.translate(...),
  [ErrorCode.unsupportedLogsFormat]: i18n.translate(...),
}
```
If any `ErrorCode` is missing TS will throw an error.

Then, we can simplify this logic with:

```suggestion
        let errorMessage = originalErrorMessage;
        const errorCode = e.body?.attributes?.errorCode;
        if (errorCode != null) {
          errorMessage = i18n.ERROR_TRANSLATION[errorCode];
        }
        setError(errorMessage);
```